src/main.rs

```diff
@@ -181,11 +181,13 @@ pub async fn do_main() -> Result<()> {
 		db.clone(),
 		ipfs.clone(),
 		cfg.max_parallel_fetch_tasks,
+		cfg.confidence,
```

Comment:
Can we put some checks on the input value? For example, if someone supplies a negative confidence. Not necessarily inside `main`.